src/cpp/dpm_sampler.cpp

```diff
@@ -390,7 +432,8 @@ int main(int argc, char** argv)
         b_alpha = dt + (eps1 * eps2 * totbases);
         b_beta = (totbases - dt) + eps2 * totbases * (1 - eps1);
 
-        theta = gsl_ran_beta(rg, b_alpha, b_beta);
+        boost::random::beta_distribution<decltype(theta)> beta(b_alpha, b_beta);
```

Comment:
why the obtuse `decltype` syntax? is it really likely that we would switch to `float` or some oddball quad-precision floating-point arithmetic?